### PR TITLE
Readme for speed benchmark

### PR DESCRIPTION
**Pull request by j-petit**
### Description
This PR adds a readme for the speed benchmark script I wrote. The PR is so big as it depends on #1572 to be merged into the feature branch first. The rendered README is best seen [here](https://github.com/j-petit/moveit/tree/readme-speed-benchmark/moveit_ros/planning/planning_components_tools) in my branch.

@BryceStevenWilley I remember that we discussed the location of this readme somewhere on github but can't find where anymore. I put it now into the `moveit_ros/planning/planning_component_tools` but I remember that you favored `moveit_core/collision_detection/`. Any additional thoughts on that?

Found our [old discussion](https://github.com/ros-planning/moveit/pull/1551#discussion_r313598512).

### CODE REVIEW

**Comment by BryceStevenWilley** [comment]
Adding what I mentioned in our sync today: the current location/content of the README is great, I think the only thing that would be worthwhile to also add to the collision_detection_bullet README is the table of collision-checking speeds.

**Comment by davetcoleman** [comment]
@j-petit I don't understand the status of these PRs - can you isolate just the README in this PR please?

**Comment by j-petit** [comment]
Done, as some of the history was squashed this PR got mixed up.

File: moveit_ros/planning/launch/collision_checker_compare.launch [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
   <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find franka_description)/robots/panda_arm_hand.urdf.xacro'" />
 
   <include file="$(find panda_moveit_config)/launch/planning_context.launch"/>
 
   <node name="compare_collision_checking_speed" pkg="moveit_ros_planning" type="moveit_compare_collision_checking_speed_fcl_bullet" respawn="false" output="screen">
     <rosparam command="load" file="$(find panda_moveit_config)/config/kinematics.yaml"/>
+    <param name="visualization" value="$(arg visualization)" type="bool"/>
   </node>
+
+  <include if="$(arg visualization)" file="$(find panda_moveit_config)/launch/demo.launch"/>
```
Comment (rhaschke) [inline moveit_ros/planning/launch/collision_checker_compare.launch]: Shouldn't you provide a specific rviz config file (to be independent from changes in panda_moveit_config)?

Comment (felixvd) [reply moveit_ros/planning/launch/collision_checker_compare.launch]: I see where you're coming from, but I think both the chance of a breaking change in that package and the impact in this one are so small that it's probably not worth it.

File: moveit_ros/planning/planning_components_tools/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Planning Component Tools
+
+## Compare collision checking speeds: FCL vs Bullet
+The launch file `collision_checker_compare.launch` starts a benchmark between FCL and Bullet as a collision detection library. The script `compare_collision_speed_checking_fcl_bullet.cpp` performs the actual speed tests.
+
+The collision scenarios can be visualized. By default, visualization is turned off in the launch file but can be activated through `visualization:=true` as an argument to the `roslaunch`. If activated, rviz is launched and visualizations like in the images below are created. For the speed test, either meshes or other available shape primitives like the box can be used.
+
+The clutter world objects are randomely placed and in such a way that they don't collide with the robot. The robot can then be moved in a colliding configuration. In the images, the red color of world objects indicates a collision with the robot.
```
Comment (felixvd) [inline moveit_ros/planning/planning_components_tools/README.md]:
```suggestion
The clutter world objects are randomly placed such that they don't collide with the robot. The robot can then be moved to a colliding configuration. In the images, the red color of world objects indicates a collision with the robot.
```
Just typo and style.

File: moveit_ros/planning/planning_components_tools/src/compare_collision_speed_checking_fcl_bullet.cpp [hunk 3]
```diff
@@ -188,8 +195,11 @@ void runCollisionDetection(unsigned int trials, const planning_scene::PlanningSc
   if (!only_self)
   {
     req.contacts = true;
-    req.max_contacts = 999;
-    req.max_contacts_per_pair = 99;
+    req.max_contacts = 99;
+    req.max_contacts_per_pair = 10;
+    // If distance is turned on, this will slow down the collision checking very much and needs reduction in the number
```
Comment (felixvd) [inline moveit_ros/planning/planning_components_tools/src/compare_collision_speed_checking_fcl_bullet.cpp]: What is meant by "this"?

Comment (j-petit) [reply moveit_ros/planning/planning_components_tools/src/compare_collision_speed_checking_fcl_bullet.cpp]: should be "it", changed.

File: moveit_ros/planning/planning_components_tools/README.md (hunk 2 above)
Comment (j-petit) [reply moveit_ros/planning/planning_components_tools/README.md]: done, thanks for catching those :)

**Comment by j-petit** [comment]
From my side yes.

**Pull request merged**